### ISSUE TITLE
Admin\Contacts - External User can edit a contact

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-registration#1607]
#### Description:
As an External User, I want the ability to add edit an existing contact

#### Acceptance Criteria:
Given I am an authenticated user with the `industry-user-admin` role (i.e., I have been Approved to have `admin` rights for an Operator)
When I navigate to the "Contacts" table
Then I can click view details on a contact row

Given that I am on a contact's details page
When I click edit
Then the fields turn editable in editing mode
https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=14631-110&t=BLKjIYcoA4IsMpx6-4

Given that I am on a contact's details page in editing mode
When I click submit
Then I can see a confirmation that this contact has been successfully edited attn: @nanyangpro 

#### Development Checklist:
- new service (or services) for editing contacts
- new api endpoint that calls the service
- service and endpoint pytest
- in FE, toggle the form to editable when clicking the edit button
- in FE, the submit button conditionally calls the create vs. update endpoint
- new vitest for FE form to confirm the correct endpoint is called and form is editable

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Dependencies
Blocked by #1351

**Comment by patrickisaac** [comment]
Dev. checklist will be added, then Backlog Refinement label can be removed

**Comment by Sepehr-Sobhani** [comment]
I wonder if we can remove the "Is this a contact in BCIERS" toggle and the "Select the user" dropdown when editing a contact. When editing a contact and populating the whole form with another user's data, we would actually be creating a new contact instead of editing the existing one. This change would also simplify development work.

**Comment by shon-button** [comment]
@patriciarussellCAS 

**Reachable at:**
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/administration/contacts/10?contacts_title=Henry%20Ives

**Comment by patriciarussellCAS** [comment]
This worked for me. However after making an edit, and submitting the change, the buttons at the bottom of the page are "Edit" and "Cancel" there is no button to press to go back to the main table. 
Will make a follow-up ticket for this.
